src/mnode/mnode-messageproc.h

```diff
@@ -41,6 +50,7 @@ class CMasternodeMessage
         READWRITE(message);
         READWRITE(sigTime);
         READWRITE(vchSig);
+        READWRITE(messageType);
```

Comment:
Need to check that it will work correct and will not crash, for tickets we have version checking to fix like case. I'm not sure how to be here. You can receive message with less size from old clients. probably add try/catch just for messageType?

Reply:
Good consideration.
Basically this messages is not used yet. And what I am doing here to to serve what you are concerning.
Frow now on, if new message needed to added, we can add the new type to the enum class, and do validation on that.

Reply:
I think you can't just add messageType at the end of serialization.
There can be old MN's that simply don't understand this (which is probably OK - they won't read messageType then) or they send a message without messageType but you are trying to read it here.
Check if we're reading from stream:
const bool bRead = ser_action == SERIALIZE_ACTION::Read;
and then check for end of stream before reading messageType:
if (!bRead || !s.eof)
     READWRITE(messageType);
else
    // set here default messageType

Reply:
I fixed as @akobrin1's suggesstion